src/clients/credentials/ClientSecret.tsx

```diff
@@ -9,42 +11,138 @@ import {
 } from "@patternfly/react-core";
 
 import type ClientRepresentation from "@keycloak/keycloak-admin-client/lib/defs/clientRepresentation";
-import type { UseFormMethods } from "react-hook-form";
 import { PasswordInput } from "../../components/password-input/PasswordInput";
 import { CopyToClipboardButton } from "../scopes/CopyToClipboardButton";
+import { useWhoAmI } from "../../context/whoami/WhoAmI";
+import { useConfirmDialog } from "../../components/confirm-dialog/ConfirmDialog";
+import { useAdminClient } from "../../context/auth/AdminClient";
+import { useAlerts } from "../../components/alert/Alerts";
 
 export type ClientSecretProps = {
+  client: ClientRepresentation;
   secret: string;
   toggle: () => void;
-  form: UseFormMethods<ClientRepresentation>;
 };
 
-export const ClientSecret = ({ secret, toggle, form }: ClientSecretProps) => {
+type SecretInputProps = Omit<ClientSecretProps, "client"> & {
+  id: string;
+  buttonLabel: string;
+};
+
+const SecretInput = ({ id, buttonLabel, secret, toggle }: SecretInputProps) => {
+  const { t } = useTranslation("clients");
+  const form = useFormContext<ClientRepresentation>();
+
+  return (
+    <Split hasGutter>
+      <SplitItem isFilled>
+        <InputGroup>
+          <PasswordInput id={id} value={secret} isReadOnly />
+          <CopyToClipboardButton
+            id={id}
+            text={secret}
+            label="clientSecret"
+            variant="control"
+          />
+        </InputGroup>
+      </SplitItem>
+      <SplitItem>
+        <Button
+          variant="secondary"
+          isDisabled={form.formState.isDirty}
+          onClick={toggle}
+        >
+          {t(buttonLabel)}
+        </Button>
+      </SplitItem>
+    </Split>
+  );
+};
+
+const ExpireDateFormatter = ({ time }: { time: number }) => {
+  const { t } = useTranslation("clients");
+  const { whoAmI } = useWhoAmI();
+  const formatter = useMemo(
+    () =>
+      new Intl.DateTimeFormat(whoAmI.getLocale(), {
+        dateStyle: "full",
+        timeStyle: "long",
+      }),
+    [whoAmI.getLocale()]
+  );
```

Comment:
I like this code! You could store the locale here to avoid duplication.
```suggestion
  const locale = whoAmI.getLocale();
  const formatter = useMemo(
    () =>
      new Intl.DateTimeFormat(locale, {
        dateStyle: "full",
        timeStyle: "long",
      }),
    [locale]
  );
```